src/compiler/frontend.rs

```diff
@@ -327,7 +338,12 @@ pub fn compile_bodyform(
                                 qq_to_expression(opts, Rc::new(quote_body))
                             } else if *atom_name == "mod".as_bytes().to_vec() {
                                 let subparse = frontend(opts, &[body.clone()])?;
-                                Ok(BodyForm::Mod(op.loc(), subparse))
+                                Ok(BodyForm::Mod(op.loc(), false, subparse))
+                            } else if *atom_name == "mod+".as_bytes().to_vec() {
```

Comment:
I suspect the `to_vec()` calls here could be a performance issue when building large programs.
iiuc, `atom_name` is a `Vec<u8>`. My understanding is that a `Vec<u8>` can be compared against a `&[u8]`, no need to allocate a new vector just to compare against.

Something like this: `*atom_name == b"mod+"`.

This is probably not for this patch though

Reply:
Overall, I do need to change this practice.  Here, I used a style similar to the rest of the code.  I am thinking about performance (and it's pretty cruddy in general).  There are larger things I did early on that can be refactored for performance as well.  In general my thought was mirroring the form of the ocaml code closely, and my rust wasn't as keen.

Reply:
I've gone in a different direction with this.  Lambdas now desugar to ordinary functions and rely on in turn treating defun functions as values.